src/llmcompressor/pytorch/model_load/helpers.py

```diff
@@ -106,6 +107,12 @@ def save_model_and_recipe(
     :param save_safetensors: whether to save as safetensors or pickle (bin)
     :param save_compressed: whether to compress sparse weights on disk
     """
+    if is_model_quantized(model):
+        from llmcompressor.transformers.sparsification.compressed_tensors_utils import (
+            modify_save_pretrained,
+        )
+
+        modify_save_pretrained(model)
```

Comment:
Why not just call the modified `save_pretrained` function directly? Doing it this way has the unintended and undocumented side effect that now the model has a modified save_pretrained function

Reply:
youll call vanilla save_pretrained if the model has not gone thru compression.
If compressed, then one of the thing we need to do is to add the quant-config, which vanilla code doesnt do

Reply:
> Why not just call the *modified* `save_pretrained` function directly

Reply:
Wrapping is not required and leads to an extra side effect.

```python3
def save_compressed(model, *args, **kwargs):
    save_fn = save_pretrained_compressed(model.save_pretrained)
    save_fn(*args, **kwargs)
```

```python3
- modify_save_pretrained(model)
- model.save_pretrained(...)
# now model has a wrapped save function as a side effect

+ save_compressed(model)
```